gazelle/python/generate.go

```diff
@@ -212,7 +212,6 @@ func (py *Python) GenerateRules(args language.GenerateArgs) language.GenerateRes
 	}
 
 	parser := newPython3Parser(args.Config.RepoRoot, args.Rel, cfg.IgnoresDependency)
-	visibility := fmt.Sprintf("//%s:__subpackages__", pythonProjectRoot)
```

Comment:
Actually, this is desirable in usecases where you have a monorepo, where you have many requirements.txt files that you don't want to mix. This change would be a regression in that front.

The `go` plugin for `gazelle` has directives to control the visibility as is documented in https://github.com/bazelbuild/bazel-gazelle, maybe introducing a directive would be useful here as well?

Reply:
With this PR you can still get the old behavior by adding this to your Python root BUILD file:
```
package(default_visibility = ["//" + package_name() + ":__subpackages__"])
```
There's some more context in [this previous PR](https://github.com/bazelbuild/rules_python/issues/1682) that accidentally included the wrong change but the right PR description.